aptos-move/move-examples/shared_account/sources/SharedAccount.move

```diff
@@ -0,0 +1,195 @@
+module SharedAccount::SharedAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+
+    // struct State records the address of the share_holder and the numerator and the denominator of a fraction
+    struct State has store {
+        share_holder: address,
+        numerator: u64,
+        denominator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: vector<State>,
+        signer_capability: Account::SignerCapability,
+    }
+
+    const EINVALID_INPUT: u64 = 0;
+    const EACCOUNT_NOT_FOUND: u64 = 1;
+    const EADDRESS_ALREADY_EXISTS: u64 = 2;
+    const EINVALID_NUMERATOR_DENOMINATOR_COMBINATIONS: u64 = 3;
+    const EINVALID_SIGNER: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+
+    fun find_greatest_common_divisor(denominator1: u64, denominator2: u64) : u64 {
+        if (denominator1 == 0) {
+            denominator2
+        }
+        else {
+            find_greatest_common_divisor(denominator2 % denominator1, denominator1)
+        }
+    }
+
+    fun find_least_common_denominator(denominator1: u64, denominator2: u64) : u64 {
+        assert!(denominator1 * denominator2 > 0, Errors::invalid_argument(EINVALID_INPUT));
+        let gcd: u64 = find_greatest_common_divisor(denominator1, denominator2);
+        assert!(gcd > 0, Errors::invalid_argument(Errors::invalid_argument(EINVALID_INPUT)));
+        (denominator1 * denominator2) / gcd
+    }
+
+    // Create and initialize a shared account
+    public fun initialize(source: &signer, seed: vector<u8>, addresses: vector<address>, numerators: vector<u64>, denominators: vector<u64>): address acquires SharedAccount {
```

Comment:
Why make things so complicated with numerators and denominators? Why just not a single array of number of shares? For example if you own 1/3 and I own 2/3, you can just pass [you, me] and [1, 2] right? The denominator is implicitly the total number of shares